rand_core/src/block.rs

```diff
@@ -372,12 +372,14 @@ where
 
     #[inline]
     fn next_u64(&mut self) -> u64 {
-        if self.index >= self.results.as_ref().len() {
+        let mut index = self.index - self.half_used as usize;
+        if index >= self.results.as_ref().len() {
```

Comment:
This change was supposed to be to `next_u32`, not `next_u64`!

Reply:
Sorry about this, I thought I changed `next_u32`. 😅 Should be fixed now.